projection.go

```diff
@@ -33,6 +40,23 @@ type ProjectionMessageHandler interface {
 	// handler. That is, the engine should call HandleEvent() with the same
 	// event message until a nil error is returned.
 	//
+	// The engine MAY provide guarantees about the order in which event messages
+	// will be passed to HandleEvent(), however in the interest of engine
+	// portability the implementation SHOULD NOT assume that HandleEvent() will
+	// be called with events in the same order that they were recorded.
+	//
+	// The key/value association and the changes to the projection state SHOULD
+	// be persisted in a single atomic operation. If the implementation is not
+	// able to guarantee such atomicity the projection state MUST be updated
+	// before the association is persisted. This guarantees that no event is
+	// ever missed because due to the association existing without the relevant
```

Comment:
```suggestion
	// ever missed due to the association existing without the relevant
```

The comment block will need to be re-flowed again after removing that word.